app/controllers/users_controller.rb

```diff
@@ -17,7 +17,25 @@ def create
   end
 
   def show
-    @user = current_user
+    if
+      @user = current_user
+    else
+      render file: 'public/404', layout: false
```

Comment:
Interesting. This looks like a security loophole that I don't really understand.